src/app.rs

```diff
@@ -29,6 +29,8 @@ impl App {
 #[derive(Serialize, Deserialize)]
 pub struct AppData {
     pub channels: StatefulList<Channel>,
+    #[serde(skip)]
+    pub messages: StatefulList<Message>,
```

Comment:
Seems to be a little bit wasteful to copy all the messages all the time. Technically, they are already stored in the channels list.

Reply:
I have reimplemented this feature so that all the vector messages stored in a channel no longer need to be copied into the messages StatefulList every time a new channel is displayed. Now each channel stores its messages in a StatefulList rather than in a simple vector. 

I also implemented mouse scrolling of the messages list, which was easy now that the channel scrolling has been merged.